### DISCUSSION TITLE
Different Way of Closing The Room

### DISCUSSION THREAD

**Discussion by ybinstock** [mozilla/hubs#3307] [General Questions & Discussion] [answered]
Hi, 

I'm trying to close the room based on a timer function instead of the click event. 

So in UI-Root.js I'm calling       this.props.hubChannel.closeHub();

Unfortunately, it's not doing anything. 

I see in hub.js 

 if (hub.entry_mode === "deny") {
      scene.emit("hub_closed");
    }

Which then does this

scene.exitVR();
    entryManager.exitScene("closed");

But my code isn't reaching that. 

I also see that there is some relationship between hub and hub-channel through HubChannel which might be how the information to close the room is passed and through phoenix-utils. 

I'm wondering if there's an easier way for me to close the room either directly UI-Root.js or to connect UI-Root to Hub.js so I can programmatically close the room. 

Thanks

**Comment by brianpeiris** [comment]
Does `closeHub` throw an error or return a value when you try it? The only reasons I could see it failing is if you don't actually have permission to close the room, in which case it would return "unauthorized", or the hubChannel socket connection has somehow closed by the time you try to use `closeHub`.

**Reply by bemyevent** [reply]
it will be good to have this feature , or maybe we could close a room without being on it , just in admin  ?

**Reply by robinkwilson** [reply]
Create an issue/feature request if you'd like that added to the admin panel.

**Comment by robinkwilson** [comment, marked as answer]
Related discussion with description of how this would work with the new API: https://github.com/mozilla/hubs/discussions/3757

**Reply by robinkwilson** [reply]
Marking as answered for now, let us know if you have any additional specific questions then "unmark" to get it back in the pipeline! 

Get excited about the new hubs API! It's going to make scripting much easier for room management and other actions!
